packages/grid/x-data-grid/src/internals/utils/cleanupTracking/TimerBasedCleanupTracking.ts

```diff
@@ -4,24 +4,37 @@ import { CleanupTracking, UnregisterToken, UnsubscribeFn } from './CleanupTracki
 const CLEANUP_TIMER_LOOP_MILLIS = 1000;
 
 export class TimerBasedCleanupTracking implements CleanupTracking {
-  timeouts = new Map<number, NodeJS.Timeout>();
+  timeouts? = new Map<number, NodeJS.Timeout>();
```

Comment:
What does this `?` means? Never encountered such a character before an `=`

Reply:
`?` is used to tell that the attribute can be `undefined`. Using it is the same as:

```tsx
export class TimerBasedCleanupTracking implements CleanupTracking {
  timeouts: Map<number, NodeJS.Timeout> | undefined = new Map<number, NodeJS.Timeout>();
}
```

Only classes and interfaces allow it. More information: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#optional-class-properties